packages/tokens-studio-for-figma/src/plugin/setValuesOnVariable.ts

```diff
@@ -7,13 +7,100 @@ import { convertTokenTypeToVariableType } from '@/utils/convertTokenTypeToVariab
 import { checkCanReferenceVariable } from '@/utils/alias/checkCanReferenceVariable';
 import { TokenTypes } from '@/constants/TokenTypes';
 import { transformValue } from './helpers';
+import { ProgressTracker } from './ProgressTracker';
+import { BackgroundJobs } from '@/constants/BackgroundJobs';
+import { notifyUI, postToUI } from './notifiers';
+import { MessageFromPluginTypes } from '@/types/messages';
 
 export type ReferenceVariableType = {
   variable: Variable;
   modeId: string;
   referenceVariable: string;
 };
 
+// Helper function to process tokens in batches to prevent memory leaks
+async function processBatch(
+  batch: SingleToken<true, { path: string, variableId: string }>[],
+  variablesInFigma: Variable[],
+  collection: VariableCollection,
+  mode: string,
+  baseFontSize: string,
+  shouldRename: boolean,
+  variableKeyMap: Record<string, string>,
+  referenceVariableCandidates: ReferenceVariableType[],
+  renamedVariableKeys: string[],
+) {
+  await Promise.all(batch.map(async (token) => {
```

Comment:
```suggestion
  return Promise.all(batch.map(async (token) => {
```
since we don't need to await for anything inside the scope of this function, we can just return it, which returns the Promise that's a result of Promise.all()